sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -38,16 +38,25 @@ class GpuProjectExecMeta(
     proj: ProjectExec,
     conf: RapidsConf,
     p: Option[RapidsMeta[_, _, _]],
-    r: DataFromReplacementRule) extends SparkPlanMeta[ProjectExec](proj, conf, p, r) {
+    r: DataFromReplacementRule) extends SparkPlanMeta[ProjectExec](proj, conf, p, r)
+    with Logging {
   override def convertToGpu(): GpuExec = {
     // Force list to avoid recursive Java serialization of lazy list Seq implementation
     val gpuExprs = childExprs.map(_.convertToGpu()).toList
     val gpuChild = childPlans.head.convertIfNeeded()
     if (conf.isProjectAstEnabled) {
-      childExprs.foreach(_.tagForAst())
       if (childExprs.forall(_.canThisBeAst)) {
         return GpuProjectAstExec(gpuExprs, gpuChild)
       }
+      // explain AST because this is optional and it is sometimes hard to debug
+      val exp = conf.explain
+      if (!exp.equalsIgnoreCase("NONE")) {
```

Comment:
Nit: We should probably put the logic for interpreting the various values of the `explain` conf in `RapidsConf` as separate, parameterless predicate functions rather than pepper the logic everywhere the conf is used.  Might be good for a followup issue.